### ISSUE TITLE
React component not detected when subclass

### ISSUE THREAD

**Issue by stianlp** [yannickcr/eslint-plugin-react#806]
Hi!

It looks like my component is not detected, when I am subclassing a class that extends React.Component.

I have a class A extends React.Component {}. 

Eslint detects this class as a react component.

But for class B extends A {}, eslint does not detect B.

Does anyone know how to fix this?

**Comment by ljharb** [comment]
There's not really a way for linter rules to know what your other files are doing. (Fwiw, React generally discourages inheriting from components, and encourages using composition to share behavior)

I believe you can use a jsdoc comment to inform this plugin that your component inherits from a react component but I don't recall the syntax.

**Comment by petersendidit** [comment]
If the code uses the JSDoc  `@extends` comment with `React.Component` or `React.PureComponent` [it will pick up that its a component](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/util/Components.js#L183-L200).
```jsx
/** @extends React.Component */
class Hello extends ChildComponent {
  render() {
    return <div>Hello</div>;
  }
}
```
